crypto.go

```diff
@@ -237,25 +251,50 @@ func (s ecdhScheme) ID() KEMID {
 	panic(fmt.Sprintf("Unsupported curve: %s", s.curve.Params().Name))
 }
 
-func (s ecdhScheme) GenerateKeyPair(rand io.Reader) (KEMPrivateKey, KEMPublicKey, error) {
-	d, x, y, err := elliptic.GenerateKey(s.curve, rand)
-	if err != nil {
-		return nil, nil, err
+func (s ecdhScheme) privateKeyBitmask() uint8 {
+	switch s.curve.Params().Name {
+	case "P-256":
+		return 0xFF
+	case "P-521":
+		return 0x01
+	}
+	panic(fmt.Sprintf("Unsupported curve: %s", s.curve.Params().Name))
+}
+
+func (s ecdhScheme) DeriveKeyPair(ikm []byte) (KEMPrivateKey, KEMPublicKey, error) {
+	idBuffer := make([]byte, 2)
+	binary.BigEndian.PutUint16(idBuffer, uint16(s.ID()))
+
+	dkp_prk := s.KDF.LabeledExtract(nil, string(idBuffer)+"dkp_prk", ikm)
```

Comment:
The `nil` salt translates to `make([]byte, s.hash.Size())` over in ecdhScheme's definition, but the spec says the salt should be `zero(0)`. Is this a discrepancy?

Reply:
Nice eye! This isn't a problem, as that behavior part of the HKDF spec (https://tools.ietf.org/html/rfc5869, 2.2).